### PR TITLE
Update z/TPF code cache strategy

### PR DESCRIPTION
**Pull request by jjohnst-us**
Pull request represents z/TPF OS platform specific updates to code cache strategy.

z/TPF has a second 64-bit heap region managed by the mmap function which allows Java processes to work better with traditional non-java applications. The update enables processes to detect whether the feature is enabled and to allocate the JIT code cache from it based on a heuristic meaningful only to z/TPF.  The fallback is to use the other pre-existing 64-bit heap reagion limited by a MAXXMMES system heap configuration unique to the z/TPF platform.

### CODE REVIEW

**Comment by jjohnst-us** [comment]
If it makes it easier, all the updates are within z/TPF guard macros and have been running for 2 years.

**Comment by jjohnst-us** [comment]
@0xdaryl 
Hey Daryl, just my periodic shoulder tap, when you get a chance to take a look at this PR.

Much appreciated.

**Comment by 0xdaryl** [comment]
Jenkins test sanity xlinux jdk17

Changes are guarded with ifdefs that OpenJ9 testing does not enable so light sanity testing seems sufficient.

**Comment by 0xdaryl** [comment]
@mpirvu : this PR introduces changes to the zTPF code cache allocation that are all guarded behind zTPF ifdefs.  In spite of that, do you have any comments you want to make prior to merging?

File: runtime/compiler/control/rossa.cpp [hunk 1]
```diff
@@ -1236,19 +1236,42 @@ onLoadInternal(
 #endif
 
 #if defined(J9ZTPF)
+
+#if defined(J9VM_OPT_JITSERVER)
+   if (!(javaVM->internalVMFunctions->isJITServerEnabled(javaVM)))
+      {
```
Comment (mpirvu) [inline runtime/compiler/control/rossa.cpp:1242]: We can protect with `ifdef` only the `if` statement, without the curly bracket. Then we don't need to protect the ending curly bracked with another `ifdef`

**Comment by jjohnst-us** [comment]
@mpirvu @0xdaryl 

Requested changes applied, is there anything else?  Thanks!

**Comment by 0xdaryl** [comment]
@mpirvu : if you approve, I'll launch some light testing on this.

File: runtime/compiler/control/rossa.cpp [hunk 2]
```diff
@@ -1236,19 +1236,42 @@ onLoadInternal(
...
+
+         void *maxmmapReturnValue;
+         uint16_t physMemory;
+
+         /* We can move this define to a header (i.e., bits/mman.h) later on */
+         void *checkSupport = mmap(NULL, 0, 0, MAP_SUPPORTED|MAP_PRIVATE, 0, 0);
+
+         /* Retrieve the maximum number of 1MB frames allocated for the z/TPF Java Process (from MAXXMMES or MAXMMAP)  */
+         /* then take 10% of that value to use for the code cache size. */
+         if (checkSupport != MAP_FAILED)  {
```
[the remark this answers is not in the document]
Comment (jjohnst-us) [reply runtime/compiler/control/rossa.cpp]: @mpirvu 

Thank you. I fixed the formatting for the curly brackets.  I also fixed the alignment of a few lines.

Yes our mmap is non-standard, over time we will be able to catch up to the standard interface, but currently it is a very low priority objective.

**Review by mpirvu — APPROVED** [review]
LGTM

**Comment by 0xdaryl** [comment]
Jenkins test sanity zlinux

**Comment by 0xdaryl** [comment]
Jenkins test sanity zlinux jdk17

**Review by 0xdaryl — APPROVED** [review]

**Pull request merged**